nixos/modules/services/cluster/kubernetes/default.nix

```diff
@@ -73,6 +73,18 @@ let
     };
   };
 
+  mkWaitCurl = { address ? cfg.apiserverAddress, sleep ? 2, path ? "", args ? "-o /dev/null",
```

Comment:
Just noticed by accident, when deploying a different deployment with nixops, nix complains about `cfg.apiserverAddress` being used but not defined. Probably because it has no default value in the options. Have to workaround that somehow.

Reply:
@calbrecht  `apiserverAddress` is mkDefault'et to the value of `masterAddress` here: https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/cluster/kubernetes/default.nix#L286 .. The reason is the assumption that you have only one master node, but it should still be possible to override this. In our production cluster for example; we have 3 master nodes and load-balanced address that divides requests between them all. In that case, we want a setup where `masterAddress != apiserverAddress`.

Reply:
@johanot Ok, so you have the `cfssl` service running on the `masterAddress`, right?

However, the issue here was that i defined `kube-control-plane-online` service and target within apiserver.nix but outside of the mkIf cfg.enabled set, which led to evaluation of this function declaration in every other nix-build, probably. Fixed this in https://github.com/NixOS/nixpkgs/pull/56789/commits/f5cacee01e68da733813b87b7d03c5beec63c430